nusight2/build_scripts/build.ts

```diff
@@ -15,6 +13,23 @@ const opts: ConfigOptions = {
   sourceMap: isContinuousIntegration ? false : 'source-map',
   rootDir: path.join(__dirname, '..'),
 }
-webpack([getClientConfig(opts), getServerConfig(opts)]).run((err, stats) => {
-  stats && process.stdout.write(stats.toString({ colors: true }) + '\n')
+
+const compiler = webpack([getClientConfig(opts), getServerConfig(opts)])
+compiler.run((err, stats) => {
+  if (err) {
+    process.stderr.write(err.message)
+    if (err.stack) {
+      process.stderr.write(err.stack)
+    }
+    process.exitCode = 1
+    return
+  }
+  if (stats) {
+    process.stdout.write(stats.toString({ colors: true }) + '\n')
+    if (stats.hasErrors()) {
+      process.exitCode = 1
+    }
+  }
+  // Calling close() allows low-priority work (like persistent caching) the opportunity to complete
+  compiler.close((closeErr) => {})
```

Comment:
Is this a variable? It is not assigned anywhere

Reply:
As in, is `compiler` a variable?

Reply:
I was basically trying to follow these docs:

https://webpack.js.org/api/node/#error-handling

https://webpack.js.org/api/node/#statshaserrors

And joey's comment, regarding `close`

https://nubotsteam.slack.com/archives/C3T1J8SGL/p1629878566062600

Reply:
Sorry should have been more specific, I don't see where `closeErr` gets it's value, I'm assuming it is a variable

Reply:
It's an anonymous function/callback. That's the parameter for it. Basically when you call close you give it that function. It will then call that function with the parameter it uses for the callback. So that's the internal object it passes to the callback when it executes it.

Reply:
In this case, i did not know of anything specific which needed to be done in it, so I did nothing.

Reply:
You should probably write it to `stderr`. We'd want to know if there was an error closing the compiler.